### PR TITLE
[KP] Add elementwise add xpu after phi, test=develop

### PR DESCRIPTION
**Pull request by Liu-xiandong**
### PR types
New features

### PR changes

OPs
### Describe

Added kernel for XPU2 KP system, elementwise_add

This PR involves 4 changes:

- Improve the XPU2 source code compilation process, modify cmake/operators.cmake and cmake/xpu_kp.cmake to support .kps files.

- Added the elementwise_add.kps file and added a  test for it. This file provides the forward XPU2 source code compilation code of elementwise_add

- Modified the relevant components of KP, compute_primitives_xpu2.h, etc., and solved some previous bugs

- In order to support source code compilation of XPU2, some underlying codes such as elementwise and device have been modified, such as elementwise_base.h and cuda_kernel_config.h, etc.

The result is as follows:
image

The KP_flags turns on:
image

### CODE REVIEW

File: cmake/xpu_kp.cmake [hunk 1]
```diff
@@ -127,9 +127,11 @@ macro(compile_kernel COMPILE_ARGS)
       kernel_build/${kernel_name}.bin.o
     COMMAND
       ${CMAKE_COMMAND} -E make_directory kernel_build
+    COMMAND
+      cp ${kernel_path}/${kernel_name}.kps kernel_build/${kernel_name}.xpu
```
Comment (AnnaTrainingG) [inline cmake/xpu_kp.cmake:131]: GPU环境下是否需要cp 成.cu文件呢

File: paddle/fluid/operators/elementwise/elementwise_add_op.kps [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+}  // namespace operators
+}  // namespace paddle
+
+namespace ops = paddle::operators;
+namespace plat = paddle::platform;
+
+REGISTER_OP_KERNEL(elementwise_add, KP, plat::XPUPlace,
+                   ops::ElementwiseAddXPUKPKernel<float>);
+
+REGISTER_OP_KERNEL(elementwise_add_grad, KP, plat::XPUPlace,
```
Comment (AnnaTrainingG) [inline paddle/fluid/operators/elementwise/elementwise_add_op.kps:185]: 此处只有XPU 注册，那GPU注册添加到哪里呢

File: paddle/phi/kernels/primitive/kernel_primitives.h [hunk 3]
```diff
@@ -14,18 +14,19 @@
...
-#include "paddle/phi/kernels/primitive/functor_primitives_xpu2.h"
+#ifdef PADDLE_WITH_XPU_KP
 
 #define KPStream XPUStream
 #define KPDevice phi::XPUContext
 #define _ptr_ _global_ptr_
 #define __forceinline__ __inline__
 #define __restrict__
 
+#include "paddle/phi/backends/xpu/xpu_context.h"
```
Comment (AnnaTrainingG) [inline paddle/phi/kernels/primitive/kernel_primitives.h:25]: 这个顺序还是有影响是吗

**Review by AnnaTrainingG — APPROVED** [review]
LGTM

File: cmake/xpu_kp.cmake (hunk 1 above)
Comment (Liu-xiandong) [reply cmake/xpu_kp.cmake:131]: 是的，目前直接转成.cu文件还有一些工作量，后续提PR完成

File: paddle/fluid/operators/elementwise/elementwise_add_op.kps (hunk 2 above)
Comment (Liu-xiandong) [reply paddle/fluid/operators/elementwise/elementwise_add_op.kps:185]: 后续会增加GPU注册，通过宏定义隔离

File: paddle/phi/kernels/primitive/kernel_primitives.h (hunk 3 above)
Comment (Liu-xiandong) [reply paddle/phi/kernels/primitive/kernel_primitives.h:25]: 有影响的

File: cmake/xpu_kp.cmake [hunk 4]
```diff
@@ -17,7 +17,7 @@ if(NOT WITH_XPU_KP)
 endif()
 
 if(NOT XPU_TOOLCHAIN)
-  set(XPU_TOOLCHAIN /workspace/paddle/xpu-demo/XTDK)
+  set(XPU_TOOLCHAIN /workspace/output/XTDK-ubuntu_x86_64)
```
Comment (xingfeng01) [inline cmake/xpu_kp.cmake:20]: 后续需要处理非x86情况

**Review by xingfeng01 — APPROVED** [review]

File: cmake/xpu_kp.cmake (hunk 4 above)
Comment (Liu-xiandong) [reply cmake/xpu_kp.cmake:20]: 好的

**Review by qili93 — APPROVED** [review]
LGTM for unittest.skipIf and skip_check_grad_ci on XPU2

**Review by ZzSean — APPROVED** [review]
LGTM for op benchmark

**Pull request merged**